localstripe/resources.py

```diff
@@ -1045,7 +1048,16 @@ def __init__(self, id=None, metadata=None, amount=None, product=None,
         try:
             assert type(id) is str and id
             assert type(active) is bool
-            assert type(amount) is int and amount >= 0
+            assert billing_scheme in ['per_unit', 'tiered']
+            if billing_scheme == 'per_unit':
+                assert type(amount) is int and amount >= 0
+            else:
+                assert tiers_mode in ['graduated', 'volume']
+                assert type(tiers) is dict and len(tiers) > 0
+                assert all(
+                    'up_to' in t and
+                    (t['up_to'] == 'inf' or t['up_to'].isdigit())
```

Comment:
Try `up_to: null`: it will crash the server (HTTP 500) instead of returning a HTTP 400.

Instead of `isdigit()`, you can use `type(t['up_to']) is int`.